usecases/evaluate_scenario/evaluate_scenario.go

```diff
@@ -185,9 +185,11 @@ func evalScenarioRule(
 			fmt.Sprintf("error while evaluating rule %s (%s)", rule.Name, rule.Id))
 	}
 
+	ruleEvaluationDto := ast.AdaptNodeEvaluationDto(ruleEvaluation)
 	ruleExecution := models.RuleExecution{
-		Rule:   rule,
-		Result: ruleReturnValue.ReturnValue,
+		Rule:       rule,
+		Evaluation: &ruleEvaluationDto,
+		Result:     returnValue,
```

Comment:
Adapt to `NodeEvaluationDto` so we manipulate the same `models.RuleExecution` everywhere